### PR TITLE
Legion core nerf - legion cores provide reduced damage slowdown on station vs immunity and are slower to apply (when on station) (not on lavaland)

### PR DESCRIPTION
**Pull request by SomeguyManperson**
# Document the changes in your pull request

Legion cores are an item that allows someone to quickly heal on their own, providing a safety net when fighting megafauna or to escape being beaten to death by a goliath after eating a tentacle. This is stuff that can otherwise feel like cheap bullshit to die to but when fighting actual people the cheap bullshit is called "tech" and "robust" and "departments that don't get nerfed", and having a cheap instant heal is now the actual cheap bullshit, especially when you can carry 20 (low average bound) of them and apply them all in under 5 seconds
To remedy this, I have removed the "instant" from "cheap instant heal", requiring legion cores to be channeled for 4 seconds to use (time based on medical items) if on the station or transit z level (not space z levels, lavaland z levels, or centcomm z levels because i hate admins 😠)

Additionally, the free morphine effect with 0 downsides has been replaced with a free nuka cola effect with 0 downsides, being while on station regenerative cores apply slowdown reduction rather than slowdown removal for their duration. This allows them to keep this gimmick while making damage to the user no longer seem completely pointless as they can just bugger off at full speed when scratched

Also replaces the duplicate code for attackself with just calling the original healing code and telling it to do its job and causes preserved cores to name themselves "preserved regenerative core" because having an active core named "decayed regenerative core" made me upset

closes #15758 

# Wiki Documentation

Legion cores now have a channel of 4 seconds / 2 seconds when applied to self/other on station or in transit
Legion cores now give reduced damage slowdown rather than damage slowdown immunity when applied on station
preserved cores are now named "preserved regenerative core", this is unlikely to be or need documentation

# Changelog

:cl:  
tweak: legion cores now have a 4 second channel for being applied on station, 2 seconds if applying to another person
tweak: legion cores now provide damage slowdown reduction rather than immunity when applied on station
rscadd: legion cores that have been preserved after decaying will no longer be called "decayed regenerative core" because preserving cores now renames them to "preserved regenerative core"
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
good alternative happy with either

File: code/modules/mining/equipment/regenerative_core.dm [hunk 1]
```diff
@@ -117,7 +130,7 @@
 
 /*************************Legion core********************/
 /obj/item/organ/regenerative_core/legion
-	desc = "A strange rock that crackles with power. It can be used to heal completely, but it will rapidly decay into uselessness."
+	desc = "A strange rock that crackles with power. It can be used to heal quickly, but it will rapidly decay into uselessness. Radiation found in active space installments will slow its healing effects.""
```
Comment (VaelophisNyx) [inline code/modules/mining/equipment/regenerative_core.dm:133]: desc = "A strange rock that crackles with power. It can be used to heal quickly, but it will rapidly decay into uselessness. Radiation found in active space installments will slow its healing effects."" <---- your rogue quote

**Comment by Skrem7** [comment]
probably the better healthier alternative bcuz instant heals are still instant heals

once it builds properly i can close mine

**Comment by Skrem7** [comment]
Check your visible_messages but otherwise looks fine I'll close mine tomorrow

File: code/modules/mining/equipment/regenerative_core.dm [hunk 2]
```diff
@@ -91,10 +82,32 @@
...
+				else
+					H.visible_message(span_notice("[user] holds [src] against [H]'s body, coaxing the regenerating tendrils from [src]..."))
+					balloon_alert(user, "Applying core...")
+					if(!do_mob(user, H, 2 SECONDS)) //come on teamwork bonus?
+						to_chat(user, span_warning("You are interrupted, causing [src]'s tendrils to retreat back into its form."))
+						return
+					balloon_alert(user, "Core applied!")
+					H.visible_message(span_notice("[src] explodes into a flurry of tendrils, rapidly covering and reinforcing [H]'s body."))
+			
+				H.visible_message(span_notice("[user] forces [H] to apply [src]... Black tendrils entangle and reinforce [H.p_them()]!"))
```
Comment (Skrem7) [inline code/modules/mining/equipment/regenerative_core.dm]: Duplicate message here

File: code/modules/mining/equipment/regenerative_core.dm [hunk 3]
```diff
@@ -91,10 +82,32 @@
...
+				if(!is_station_level(get_turf(user)) || is_reserved_level(get_turf(user)))
+					to_chat(user, span_notice("You crush [src] within your hand. Disgusting tendrils spread across your body, hold you together and allow you to keep moving, but for how long?"))
+				else
+					to_chat(user, span_notice("You hold [src] against your body, coaxing the regenerating tendrils from [src]..."))
+					balloon_alert(user, "Applying core...")
+					if(!do_after(user, 4 SECONDS, src))
+						to_chat(user, span_warning("You are interrupted, causing [src]'s tendrils to retreat back into its form."))
+						return
+					balloon_alert(user, "Core applied!")
+					to_chat(user, span_notice("[src] explodes into a flurry of tendrils, rapidly spreading across your body. They will hold you together and allow you to keep moving, but for how long?"))
```
Comment (Skrem7) [inline code/modules/mining/equipment/regenerative_core.dm:108]: Don't you think it would be better if a self-applied legion core had some external tell for anybody watching?

**Comment by TheRyeGuyWhoWillNowDie** [comment]
at this point just remove legion cores and make damage slowdown not apply on lavaland instead of making them this useless

**Comment by ToasterBiome** [comment]
> at this point just remove legion cores and make damage slowdown not apply on lavaland instead of making them this useless

useless = useful for their intended purpose (lavaland)?

**Comment by TheRyeGuyWhoWillNowDie** [comment]
> > at this point just remove legion cores and make damage slowdown not apply on lavaland instead of making them this useless
> 
> useless = useful for their intended purpose (lavaland)?

just spawn miners with lavaland exile implants at this point if them having literally anything worthwhile scares you that much

**Comment by ToasterBiome** [comment]
> > > at this point just remove legion cores and make damage slowdown not apply on lavaland instead of making them this useless
> > 
> > 
> > useless = useful for their intended purpose (lavaland)?
> 
> just spawn miners with lavaland exile implants at this point if them having literally anything worthwhile scares you that much

lets do it

**Comment by SomeguyManperson** [comment]
> at this point just remove legion cores and make damage slowdown not apply on lavaland instead of making them this useless

would you like to explain your reasoning or are you just going to call any change you don't make turning something "useless" or "overpowered"

**Comment by TheRyeGuyWhoWillNowDie** [comment]
> would you like to explain your reasoning or are you just going to call any change you don't make turning something "useless" or "overpowered"

ok

legion cores are decently rare most of the time for miners (unless they find a legion tendril, but i would be fine with those being removed/having the legion core drops from them removed because they turn legion cores, a limited and quite powerful resource into something you can farm) 

turning them into a joke that takes ass loads of time to use for the effect to be bad (only against stun weapons now that damage slowdown only triggers for stun weapons... hmm...) and making them have a time you have to stand still during to apply for this reduced effect invalidates them on station, 'b-but miners aren't supposed to have loot!' then what the fuck is the point of fighting fauna? if the rewards are constantly being removed/reduced, why hasn't the:

Instant round removal holes been removed
Chainstun fauna been removed
Watchers that fire slowdown beams been removed

all of these gradual hugbox nerfs for mining make them weaker than LITERALLY ANY OTHER JOB, and one that they constantly have to put themselves at risk of round removal for. The only legitimate rewards that miners get for fighting fauna (that round removes you if you fail) is voice of god and HECK suit, but I know for a fact someone is going to effectively remove heck suit very shortly. Survival pens will probably be next on the chopping block, because miners having any survivability on the space station is bad, apparently. Botany gets potency 100 coniine death nettles, Chemistry gets larger-than-maxcap nitroglycerin grenades, Security has access to print infinite guns for very little cost. Shaft miners get armor that randomly stuns you from time to time, and a worse combat knife. 

tl;dr stop nerfing shaft miners without nerfing lavaland to be equivalent to what modern day mining rewards can get for you

**Comment by boodaliboo** [comment]
this will likely get merged but i think this pr is a bit ridiculous if someone goes through the effort of having legion cores on the station especially if there stabilized, they should be able to use them, in all my 1.5k hours playing i have never seen them abused on the station

**Review by JamieD1 — APPROVED** [review]
